tests/type_inference/type_inference_visitor_unittest.py

```diff
@@ -0,0 +1,305 @@
+"""Unittest for the type_inference_visitor."""
+
+import unittest
+from astroid.transforms import TransformVisitor
+import astroid
+from type_inference_visitor import *
+
+type_visitor = TransformVisitor()
+type_visitor.register_transform(astroid.Const, set_const_type_constraints)
+type_visitor.register_transform(astroid.Tuple, set_tuple_type_constraints)
+type_visitor.register_transform(astroid.List, set_list_type_constraints)
+type_visitor.register_transform(astroid.Dict, set_dict_type_constraints)
+type_visitor.register_transform(astroid.UnaryOp, set_unaryop_type_constraints)
+type_visitor.register_transform(astroid.BinOp, set_binop_type_constraints)
+
+
+class SetConstFunctionTest(unittest.TestCase):
+    """testers specifically for the function set_const using single nodes."""
+
+    def test_const_str(self):
+        """testing if the function set_const has the correct output for node
+        type str."""
+        # can't work without assigning, string will be evaluated to a Name Obj.
+        module = astroid.parse("""a='sample_string'""")
+        type_visitor.visit(module)
+        expected = str
+        # module is a module() object, module.body is a list object,
+        # module.body[0] is an Assign object, module.body[0].value is a
+        # Const(str) node.
+        self.assertEqual(expected, module.body[0].value.type_constraints)
```

Comment:
Did we not talk about using the `nodes_of_class` method? Did that fail for some reason I'm forgetting?